deploifai/api/__init__.py

```diff
@@ -410,3 +409,213 @@ def create_project(self, project_name: str, cloud_profile: CloudProfile, fragmen
+        }
+
+        try:
+            r = requests.post(
+                self.uri,
+                json={"query": query, "variables": variables},
+                headers=self.headers,
+            )
+            api_data = r.json()
+
+            return api_data["data"]["trainingInfrastructurePlans"]
+        except TypeError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+        except KeyError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+
+    def falcon_ml_config(self, variable):
+        query = (
+            """    
+            query ($where: FalconMLConfigWhereInput)
+                {
+                    falconMLConfigs(where: $where){
+                        id
+                        pythonVersion
+                        frameworkVersion
+                        cudaVersion
+                        cudnnVersion
+                    }
+                }
+            """
+        )
+        variables = {
+            "where": variable
+        }
+
+        try:
+            r = requests.post(
+                self.uri,
+                json={"query": query, "variables": variables},
+                headers=self.headers,
+            )
+            api_data = r.json()
+
+            return api_data["data"]["falconMLConfigs"]
+        except TypeError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+        except KeyError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+
+    def falcon_ml_config_distinct(self, variable, distinct_option: str):
+        query = (
+            """    
+            query ($where: FalconMLConfigWhereInput $distinct: [FalconMLConfigScalarFieldEnum!])
+                {
+                    falconMLConfigs(where: $where distinct: $distinct){
+                        id
+                        pythonVersion
+                        frameworkVersion
+                        cudaVersion
+                        cudnnVersion
+                    }
+                }
+            """
+        )
+        variables = {
+            "where": variable,
+            "distinct": [distinct_option],
+        }
+
+        try:
+            r = requests.post(
+                self.uri,
+                json={"query": query, "variables": variables},
+                headers=self.headers,
+            )
+            api_data = r.json()
+
+            return api_data["data"]["falconMLConfigs"]
+        except TypeError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+        except KeyError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+
+    def create_training_server(self, name: str, project_id: str, cloud_profile_id: str,
+                               falcon_plan: str, falcon_gpu: str,
+                               data_storage_id: str = None, falcon_id: str = None):
```

Comment:
isn't it supposed to be `data_storage_ids` of `List[str]` type? This is eventually passed as `dataStorageIds` variable in graphql, right?

Reply:
yes
but it also accepts a singular string as input
But list[None] does not work which is the way it works in case the user wants to not use a data storage